# Formwork for Slabs

<iframe width="420" height="315" src="//www.youtube.com/embed/y90s_JJmpQU" frameborder="0" allowfullscreen></iframe>

# Slab Forms

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Slab%20Forms.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

<a href = "http://i.imgur.com/dsYZ9Pt.png"><img src="http://i.imgur.com/dsYZ9Pt.png" alt = "Slab Form Parts" title = "Slab Form Parts" style="max-width:300px; max-height:300px; border:1px solid blue; float:right; margin-left:10px;"/></a>

A slab form, at its most basic, consists of four parts:

* Plywood, that supports the slab
* Joists, that support the plywood
* Stringers, that support the joists
* Shores, that support the stringers

In a real slab form there are at least two other parts, the braces, and the form work for the sides of the slab.  You already know about the braces, and you can think of the [side forms](http://i.imgur.com/kCu8aMV.png) as just really short walls that are supported with nails at the bottom, and compression braces at the top.

<a href = "https://ellismanufacturing.com/collections/all/products/shore-clamps"><img src="http://i.imgur.com/OgBEziC.jpg" alt = "Slab Form and Reshores" title = "Slab Form and Reshores" style="max-width:300px; max-height:300px; border:1px solid blue; float:left; margin-right:25px;"/></a>


In the image to the right, the shores are green due to being pressure treated.  This is not necessary, but I've shown them this way because if you walk into a home center you are most likely to see 4x4s that are green due to being treated.

The image to the left ([source](http://www.ellisok.net/gallery)) is of a third floor slab form, supported on [Ellis Jacks](http://www.ellisok.net/shore-clamps/ellis-4x4-shore-clamp), (note that there are addition eccentricities in the shores).  The second floor is being supported by reshores, but that is another topic.  Additionally, there are options for [shoring$^1$](http://www.sontac-scaffold.com/images/Products/prop-001.gif), and even [forms$^2$](http://www.peri.com/shared/siteimg/history/1992_01.jpg).  But, again, if you can understand how to design a slab form out of timber, then you can design one using other systems.

1. [Source](http://www.sontac-scaffold.com/en/products/steel-prop.htm)
2. [Source](http://www.peri.com/en/company/company/history.cfm)

## Design Process

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Design%20Process.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

**Changed**
The design procedure is similar to wall forms.

1. Determine the load on the plywood.
2. Determine the support spacing for the plywood, (joists are used for support instead of studs in this case).
3. Determine the support spacing for the joists, (stringers are used for support instead of wales).
4. Determine the support spacing for the stringers, (shores instead of ties).
5. Determine the bearing strength (one additional case in slab forms).
    1. Bearing between joists and stringers.
    2. Bearing between stringers and shores.
6. Determine the capacity of the shore, (we will use timber columns for our shores).
7. Determine the spacing of the shores.

Aside from the fact that there is one additional bearing check we need to make, there is one other difference.  That difference is the constructible spacings.  When we were looking are wall forms everything we looked at was based around getting the spacings to work out with the size of a sheet of plywood.  We only need to design around a sheet of plywood when we are doing the plywood support spacings in slab forms, i.e. the joist spacings.  Our joists are not limited to $48"$ and $96"$, instead joists will generally be a standard lumber length.  Lumber is typically sold in either $8'$, $10'$, $12'$, $14'$, or $16'$ lengths.  This variety means that we will have to decide how long we want the joists to be when we determine their support spacings.  For instance if we find that the joists should be supported every $42"$, then instead of rounding down to $32"$ to use an $8'$ long member ($3 \times 32" = 96" = 8'$) we will just use $14'$ long joists ($4 \times 42" = 168" = 14'$).  I'll cover an analytically approach of determining this later.

Before we start working on an example problem, I should mention a couple of assumptions I will be making in all slab design problems.  Firstly, all joists will be 4x4s.  This isn't required in a professional setting, but they are used commonly enough that it isn't an unreasonable assumption.  Secondly, all stingers will be 4x6s (oriented in the strong direction).  Again, it isn't really required but it is relatively common.  

The reason that 4x4s and 4x6s are *so* common that I can make these assumptions, is that the use such stout members eliminates the need for labor intensive, intermediate blocking.  Such blocking *would* be needed to prevent rollover for taller and thinner members.   Let's go ahead, and work through an example problem.

# Slab Form Design Example:

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Determine%20the%20Following.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

### Determine the Following:

**Changed**
* Joist Spacing in inches
* Stinger Spacing in inches
* Shore Spacing in inches
* Shore Axial Load in pounds
* Allowable Stringer Bearing Stress in psi
* Actual Stringer Bearing Stress in psi
* Verify that a 4x4 shoring will work.

Design formwork for a concrete floor slab based on the following information:

* Thickness of concrete slab = 10”
* Unbraced length for shores = 14 ft
* Concrete unit weight = 150 lb/ft$^3$
* Deflection limited to L/360 but not greater than 1/16”
* All form lumber will be S4S Select Structural Hem-Fir with no splits
* Assume wet conditions for lumber and wet conditions for plywood
* Decking for the slab will be 3/4” Group II Plywood (S-2)
* Load duration factor of 7 days
* Dead Load for formwork material = 5.5 psf and Live Load = 50 psf

In [1]:
%run Older_Classes/Temp_struc_v_3_beta.py
%run beams/Timber_beam_v_02.1.py
# %run beams/Find_Beam_lib.py

In [2]:
t_concrete = 10 #in
DL_form_work = 5.5 #psf
LL = 50 #psf
Wet = True

In [3]:
ply = panel()

#setting the thickness sets the structural properties 
#of the plywood automatically
ply.nom_thick='3/4' 
ply.group=2
ply.grade='S-2'
ply.condition = "wet"

In [4]:
## Changed - and propegates... ##
Species = "HEM-FIR"
Grade = 'Select Structural'
Shore_length = 14 #ft
Shore_section_b = 3.5 #in
Shore_section_d = 3.5 #in


## Span calculations

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Span%20calculations.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

For Bending, span, $L = 10.95 \sqrt{\frac{F_b' S}{w}}$

For Rolling shear in sheathing, span, $L = \frac{20 \times F_s'}{w}\times \frac{I b }{Q}$

For Shear in dimensional lumber, span, $L = 13.33 \times \frac{F_v' b d}{w} + 2 d$

For deflection where $\Delta = {}^L/_{360}$,  span, $L = 1.69 \sqrt[3]{\frac{E' I}{w}}$

For deflection where $\Delta = {}^1/_{16}\text{"}$,  span, $L = 3.23 \sqrt[4]{\frac{E' I}{w}}$

Stress = Load/Area

## Pressure

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Pressure.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [5]:
DL_fw = DL_form_work
LL = LL
t = t_concrete
SlabLoad = slab_load(t,LL = LL, DL_forms = DL_fw)
p = SlabLoad.value
# p =  DL_fw+DL_con+LL
print("Uninform pressure = %.2f psf"%p)

Uninform pressure = 180.50 psf


The load on the slab formwok is made up of, the Concrete Dead Load, the Formwork Dead Load, and the worker Live Load. We need to find the Concrete Dead Load, then calculate the total pressure $(p)$ on the slab formwork.

In [6]:
SlabLoad.work

<IPython.core.display.Latex object>

## Plywood

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Plywood.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [7]:
print('Plywood thickness = %s"'%ply.nom_thick)
print('Plywood group = %s'%ply.group)
print('Plywood stress ratting = %s'%ply.grade)
print('Plywood moisture condition = %s'%ply.condition)

Plywood thickness = 3/4"
Plywood group = 2
Plywood stress ratting = S-2
Plywood moisture condition = wet


In [8]:
# OVERRIDE
# ply.F_b = 820 #psi/ft
# ply.F_s = 44 #psi/ft
# ply.E = 1300000 #psi/ft
# ply.C_D = 1.25

Knowing that the plywood is oriented in the strong direction, and our $C_D = 1.25$, we can look up the properties of the plywood in the [Plywood Design Specification](http://www.apawood.org/SearchResults.aspx?q=Y510&tid=1).

In [9]:
ply.show_props.work

<IPython.core.display.Latex object>

This means that the allowable stresses for the plywood are,

In [10]:
ply.show_allowable_stress.work

<IPython.core.display.Latex object>

In [11]:
w = p
print("w = %.2f plf"%w)
joist_spacing = board_test(ply,w,plywood=True, slab=True)

w = 180.50 plf


## Plywood Span Limits (Joist Spacing)

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Plywood%20Span%20Limits%20(Joist%20Spacing).mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

Now that we know the properties of the plywood we can calculate the maximum spacing of the joists due to bending, rolling shear, and both deflection limits.

In [12]:
joist_spacing.work

<IPython.core.display.Latex object>

## Joist Line Load

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Joist%20Line%20Load.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [13]:
print("Using {} {}\n".format(Grade,Species))
joist=Timber_Beam() 
joist.species=Species
joist.C_M.use=Wet
joist.grade=Grade

Using Select Structural HEM-FIR

loads -- a list of load dictionaries. For example:
    
    loads=[{'x0':5,'x1':10,'w':10}, #uniformly distributed load over a 5ft section (5ft - 10ft)
           {'x0':0,'x1':5,'w':20},
           {'x0':10,'x1':15,'w':20},
           {'x0':12.5,'p':20}, #point load 
           {'x0':0,'x1':20,'arb':'1/10*x'}, #triangluar load function
           {'x0':10,'m':20}] # a concentrated moment
    It is beneficial to add the 'Source' and 'Symbol' to the load dictionary for 
    load combinations and deflection calculations.  Source would be like "Dead",
    "Live", or etc.  Symbol should be "D", "L", "L_r", "W", "E", or ect.  This 
    is not required, but it is stongly encouraged.
           
If you need to display plot you will need to used the following commmand
    %matplotlib inline
Note: The atribute 'pick_list' is now available_grades (added Jan, 6, 2014)

Now using 2015 Timber code by default
Setting u_c_f to 12**3 based on length in ft and d in inche

In [14]:
joist.C_D = 1.25
joist.C_r_setter(joist_spacing.value.inches)

In [15]:
lineload = line_load(p,joist_spacing.value.feet)
w = lineload.value

We need to fine a new value for $w$ based on the joist spacing.

In [16]:
lineload.work

<IPython.core.display.Latex object>

In [17]:
joist.b=3.5
joist.d=3.5
stringer_spacing = board_test(joist,w)

## Joist Allowable Stresses

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Joist%20Allowable%20Stresses.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

After looking up the value for $F_b$, $F_v$, and $E$ in Table 4A, and also noting the values of $C_F$, $C_M$, and $C_r$, we and go to Table 4.3.1 to determine our allowable stresses.  (Every adjustment factor we don't use equals 1.)

In [18]:
joist.F_b_latex

<IPython.core.display.Latex object>

In [19]:
joist.F_v_latex

<IPython.core.display.Latex object>

In [20]:
joist.E_latex

<IPython.core.display.Latex object>

## Joist Cross Sectional Properties

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Joist%20Cross%20Sectional%20Properties.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

From Table 1B of the Supplement we find that the sections properties of the joists are...

In [21]:
joist.geometry_latex

<IPython.core.display.Latex object>

## Joist Span Limits (Stringer Spacing)

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Joist%20Span%20Limits%20(Stringer%20Spacing).mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

Now that we know the properties of the joist, we can calculate the maximum spacing of the stringers based on bending, rolling shear, and both deflection limits.

In [22]:
stringer_spacing.work

<IPython.core.display.Latex object>

## Stringer Constructible Spacings

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Stringer%20Constructuable%20Spacings.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

We get the $4'$ spacing for the stringers based on the following procedure.  We divide all the available lumber lengths by the smallest allowable spacing, and determine which one is closest to the next whole number divisor.  Round this number up, and divide the corresponding length by that value to determine the optimum spacing.  I.e.,

* $\frac{8'}{53.42"} \times \frac{12"}{1'} = 1.80 \Leftarrow \text{The value closest to the next integer controls.}$
* $\frac{10'}{53.42"} \times \frac{12"}{1'} = 2.25$
* $\frac{12'}{53.42"} \times \frac{12"}{1'} = 2.70$
* $\frac{14'}{53.42"} \times \frac{12"}{1'} = 3.14$
* $\frac{16'}{53.42"} \times \frac{12"}{1'} = 3.59$

Rounding $1.80$ up to the next whole number we get $2$.  We now divide $8'$ by $2$ to determine the optimum spacing.

$$\frac{8'}{2} \times \frac{12"}{1'} = 48"$$

Now that we know that our spacing is $4'$ on center we can just pick the cheapest lumber size that is divisible by $4'$, e.g. either $8'$, $12'$, or $16'$ (probably $8'$).  This means that we will want to use $8'$ material for our joists.

## Stringer Line Loads

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Stringer%20Line%20Loads.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [23]:
print("Using {} {}\n".format(Grade,Species))
stringer = Timber_Beam()
stringer.species=Species
stringer.C_D=1.25
stringer.C_M.use=Wet
stringer.grade=Grade

Using Select Structural HEM-FIR

loads -- a list of load dictionaries. For example:
    
    loads=[{'x0':5,'x1':10,'w':10}, #uniformly distributed load over a 5ft section (5ft - 10ft)
           {'x0':0,'x1':5,'w':20},
           {'x0':10,'x1':15,'w':20},
           {'x0':12.5,'p':20}, #point load 
           {'x0':0,'x1':20,'arb':'1/10*x'}, #triangluar load function
           {'x0':10,'m':20}] # a concentrated moment
    It is beneficial to add the 'Source' and 'Symbol' to the load dictionary for 
    load combinations and deflection calculations.  Source would be like "Dead",
    "Live", or etc.  Symbol should be "D", "L", "L_r", "W", "E", or ect.  This 
    is not required, but it is stongly encouraged.
           
If you need to display plot you will need to used the following commmand
    %matplotlib inline
Note: The atribute 'pick_list' is now available_grades (added Jan, 6, 2014)

Now using 2015 Timber code by default
Setting u_c_f to 12**3 based on length in ft and d in inche

In [24]:
lineload = line_load(p,stringer_spacing.value.feet)
w = lineload.value

We need to fine a new value for $w$ based on the stringer spacing.

In [25]:
lineload.work

<IPython.core.display.Latex object>

In [26]:
stringer.b=3.5
stringer.d=5.5
shore_spacing = board_test(stringer,w)

## Stinger Allowable Stresses

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Stinger%20Allowable%20Stresses.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

After looking up the value for $F_b$, $F_v$, and $E$ in Table 4A (we'd use 4B for Southern Pine), and also noting the values of $C_F$, $C_M$, and $C_r$, we go to Table 4.3.1 to determine our allowable stresses.  Every adjustment factor we don't use equals 1.

In [27]:
stringer.F_b_latex

<IPython.core.display.Latex object>

In [28]:
stringer.F_v_latex

<IPython.core.display.Latex object>

In [29]:
stringer.E_latex

<IPython.core.display.Latex object>

## Stringer Cross Sectional Properties

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Stringer%20Cross%20Sectional%20Properties.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

From Table 1B of the Supplement we get the sections properties for the stringers.

In [30]:
stringer.geometry_latex

<IPython.core.display.Latex object>

## Stringer Span Limits (Shore Spacing)

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Stringer%20Span%20Limits%20(Shore%20Spacing).mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

Now that we know the properties of the stringers, we can calculate the maximum spacing of the shores based on bending, shear, and both deflection limits.

In [31]:
shore_spacing.work

<IPython.core.display.Latex object>

## Shore Constructible Spacings

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Shore%20Constructuable%20Spacings.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

We get the $56"$ spacing based on the following procedure.

* $\frac{8'}{56.97"} \times \frac{12"}{1'} = 1.69 $
* $\frac{10'}{56.97"} \times \frac{12"}{1'} = 2.21$
* $\frac{12'}{56.97"} \times \frac{12"}{1'} = 2.52$
* $\frac{14'}{56.97"} \times \frac{12"}{1'} = 2.95 \Leftarrow \text{The value closest to the next whole number controls.}$
* $\frac{16'}{56.97"} \times \frac{12"}{1'} = 3.37$

Rounding $2.95$ up to the nearest whole number we get $3$.  We now divide $14'$ by $3$ to determine the optimum spacing.

$$\frac{14'}{3} \times \frac{12"}{1'} = 56"$$

Since $14'$ is the only size that is dividable by $56"$ we should use $14'$ stringers, (unless you can get a deal, on some other length lumber that is great enough to justify a less than optimum spacing).

** Changed **

## Shore Capacity

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Shore%20Verification.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

** Changed **

In this problem, the shores are timber 4x4s.  I had you write an Excel spreadsheet to calculate the capacity of a timber column earlier this semester so you should simply calculate the capacity.  In this case I'm going to tell you that these shores will hold $2,229 lbs$.  On your own site you may have [shoring jacks](https://ellismanufacturing.com/collections/screw-jacks?gclid=Cj0KEQjwnsPGBRDo4c6RqK-Oqu8BEiQAwNviCdjYHVHd0VqudpMHZna9GkJ5lqiUUSy-zyShi5B_kCwaApi18P8HAQ) or [steel post shores](http://www.aluma.com/us/products/formwork_shoring/shoring/steel_post_shores) with known capacities, so you would just use that value.  If you want to review how to find a timber column's capacity you can view the slides below, otherwise lets find these shores' spacings.

In [32]:
print("Using {} {}\n".format(Grade,Species))
shore = Timber_Beam()
#shore.P = P #from above
shore.species=Species
shore.C_D=1.25
shore.C_M.use=Wet
shore.grade=Grade

Using Select Structural HEM-FIR

loads -- a list of load dictionaries. For example:
    
    loads=[{'x0':5,'x1':10,'w':10}, #uniformly distributed load over a 5ft section (5ft - 10ft)
           {'x0':0,'x1':5,'w':20},
           {'x0':10,'x1':15,'w':20},
           {'x0':12.5,'p':20}, #point load 
           {'x0':0,'x1':20,'arb':'1/10*x'}, #triangluar load function
           {'x0':10,'m':20}] # a concentrated moment
    It is beneficial to add the 'Source' and 'Symbol' to the load dictionary for 
    load combinations and deflection calculations.  Source would be like "Dead",
    "Live", or etc.  Symbol should be "D", "L", "L_r", "W", "E", or ect.  This 
    is not required, but it is stongly encouraged.
           
If you need to display plot you will need to used the following commmand
    %matplotlib inline
Note: The atribute 'pick_list' is now available_grades (added Jan, 6, 2014)

Now using 2015 Timber code by default
Setting u_c_f to 12**3 based on length in ft and d in inche

In [33]:
shore.b=Shore_section_b
shore.d=Shore_section_d
shore.L = Shore_length
shore.C_P_l_e.c = 0.8
print('Shore section: b = %.2f", and d = %.2f"'%(shore.b,shore.d))
print('Shore length (unbraced length) = %.2f"'%(Shore_length*12))

Shore section: b = 3.50", and d = 3.50"
Shore length (unbraced length) = 168.00"


## Shore Cross Sectional Properties

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Shore%20Cross%20Sectional%20Properties.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

From Table 1B of the Supplement we get the sections properties of the shores.

** Changed below **

In [34]:
shore.geometry_latex

<IPython.core.display.Latex object>

** Changed **

## Shore Allowable Load

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Shore%20Allowable%20Stresses.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

After looking us the value for $F_c$, and $E_{min}$ in Table 4A (we'd use 4B for Southern Pine) and also noting the values of $C_F$, and $C_M$, we and go to Table 4.3.1 to determine our allowable value for $E_{min}$.  Every adjustment factor we don't use equals 1.

In [35]:
shore.E_min_latex

<IPython.core.display.Latex object>

## $C_P$ - The Column Stability Factor

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/CP%20-%20The%20Column%20Stability%20Factor.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>


** Changed below **

In [36]:
shore.C_P.work

<IPython.core.display.Latex object>

### Allowable Compressive Stress

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Allowable%20Compressive%20Stress.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [37]:
shore.F_c_latex

<IPython.core.display.Latex object>

## Shore Load Capacity

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Shore%20Load%20Capacity.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

** Changed **

Now we determine the shore's capacity to find the allowable spacing.

** Changed below **

In [38]:
shore.column_capacity_brief.work

<IPython.core.display.Latex object>

## Actual Shore Spacings

**Need Audio**

Since the axial load on the shore due to the slab load is found by the following we can solve for the maximum shore spacing based on the shore capacity.
$$w_{stringer} = \text{Pressure}\times\text{Stringer spacing}$$

$$P_{actual} = w_{stringer}\times\text{Shore spacing}$$
$$\text{Maximum Shore spacing} = \frac{P_{allowed}}{w_{stringer}}$$
If this winds up being less than the stringer support spacing we will need to use this value instead.

In [39]:
_results = namedtuple("Results","value text work")
_Value = namedtuple("value","feet inches")

def _max_shore_spacing(p,P_allowed):
        spacing = P_allowed/lineload.value
        s = _Value(spacing,spacing*12)
        txt = r"$$\text{Shore spacing} = \frac{P_{allowed}}{w_{stringer}}$$"
        txt += "\n"
        txt += (r"$$\text{Shore spacing} = \frac{%.0f lbs}{%g plf}$$"%(P_allowed,
                                           lineload.value)+"\n")
        txt += (r"$$\text{Shore spacing} = %.2f' "%s.feet+r'\Rightarrow %.2f"$$'%s.inches)
        return _results(s,txt,Latex(txt))

max_shore_spacing = _max_shore_spacing(p,shore.column_capacity_brief.value)
        

In [40]:
max_shore_spacing.work

<IPython.core.display.Latex object>

In [41]:
l = [96,120,144,14*12,16*12]
for L in l:
    v = L/max_shore_spacing.value.inches
    print(f"{L/12:>2.0f}ft/{max_shore_spacing.value.inches:.2f}in = {v:.2f}")

 8ft/37.04in = 2.59
10ft/37.04in = 3.24
12ft/37.04in = 3.89
14ft/37.04in = 4.54
16ft/37.04in = 5.18


Since this is less than the $56"$ we found previously we will have to use this value.

* $\frac{8'}{37.04"} \times \frac{12"}{1'} = 2.59 $
* $\frac{10'}{37.04"} \times \frac{12"}{1'} = 3.24$
* $\frac{12'}{37.04"} \times \frac{12"}{1'} = 3.89 \Leftarrow \text{The value closest to the next whole number controls.}$
* $\frac{14'}{37.04"} \times \frac{12"}{1'} = 4.54$
* $\frac{16'}{37.04"} \times \frac{12"}{1'} = 5.18$

Rounding $3.89$ up to the nearest whole number we get $4$.  We now divide $12'$ by $4$ to determine the optimum spacing.

$$\frac{12'}{4} \times \frac{12"}{1'} = 36"$$

We should use $12'$ stringers with the shores spaced $36"$ on center.

In [42]:
from Older_Classes.Constructibility import constructible,ConstructQuestion
_shore_construct = constructible(max_shore_spacing.value.inches)
if _shore_construct.spacing < shore_spacing.value.inches:
    txt = ('Due to the shores being unable to carry the full load at the '
           'calculated stringer support spacings (and for '
           'constructibility reasons) the %g ft long stringers '
           'need to be supported by the shores every %g" instead of every %g".'
           %(_shore_construct.board,_shore_construct.spacing,
             shore_spacing.value.inches))
    actual_shore_spacing = _shore_construct.spacing
else:
    txt = ('Since the shores can carry more than what is required, we use the '
           'calculated stringer support spacings of %g".'
           %(shore_spacing.value.inches))
    actual_shore_spacing = shore_spacing.value.inches

s_c = _Value(_shore_construct.spacing/12,_shore_construct.spacing)
constructible_shores = _results(s_c,txt,Latex(txt))

In [43]:
constructible_shores.work

<IPython.core.display.Latex object>

** Added **

## Shore Load

In [44]:
s1 = stringer_spacing.value.inches
s2 = actual_shore_spacing
axial_load = shore_load(p,s1,s2)

In [45]:
axial_load.work

<IPython.core.display.Latex object>

## Bearing Pressure

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Bearing%20Pressure.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>


<a href = "http://i.imgur.com/lCGtXNz.png"><img src="http://i.imgur.com/lCGtXNz.png" alt = "Slab Form Bearing" title = "Slab Form Bearing" style="max-width:300px; max-height:300px; border:1px solid blue; float:right; margin-left:10px;"/></a>

We need to check the bearing pressure for both where the joist cross the stringers, and where the stingers are supported by the shores.

#### Where the Stingers support the Joists:

In [46]:
s1 = joist_spacing.value.feet
s2 = stringer_spacing.value.feet
c1 = joist.b #contact dimension
c2 = stringer.b #contact dimension
bearing_loads = bearing_info(p, s1, s2, c1, c2, 'joist width', 'stringer width')
A = bearing_loads.area
P = bearing_loads.load

We have to first find the bearing load of the joists on the stringers.

In [47]:
bearing_loads.work

<IPython.core.display.Latex object>

We will have to look up the $F_{c \perp}'$ adjustment factors in Table 4.3.1 and the value for $F_{c \perp}$ and $C_M$ in the Supplement to be able to compare $f_{c \perp}$ to $F_{c \perp}$.

In [48]:
bearing_check(stringer,P,A,board_name="stringers and joist").work

<IPython.core.display.Latex object>

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Where%20the%20Shores%20support%20the%20Stingers.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

#### Where the Shores support the Stingers:

In [49]:
s1 = actual_shore_spacing/12
s2 = stringer_spacing.value.feet
c1 = Shore_section_b #contact dimension
c2 = Shore_section_d #in #contact dimension
bearing_loads = bearing_info(p, s1, s2, c1, c2, 'shore width', "shore depth")
A = bearing_loads.area
P = bearing_loads.load

Now we find the bearing load of the stringers on the shores.

**Change below**

In [50]:
bearing_loads.work

<IPython.core.display.Latex object>

The values for $F_{c \perp}$ and $C_M$ are the same as before.  We just need to recalculate $f_{c \perp}$ to be able to compare it to $F_{c \perp}$.

** Change below **

In [51]:
bearing_check(stringer,P,A,board_name="stringer").work

<IPython.core.display.Latex object>

## Summary

<audio controls data-autoplay preload>
 <source src="https://github.com/damontallen/Construction-Lectures-Fall-2014/raw/master/Lecture%20Audio/Week%209/Slab%20Form%20Design/Summary.mp3"
         type='audio/mp3'>
 <p>Your user agent does not support the HTML5 Audio element or the files are missing.</p>
</audio>

In [52]:
txt = (r'$$\text{Using %s", Group %s, %s stress rated, plywood, '
       %(ply.nom_thick, ply.group, ply.grade))
txt += r'supporting a %g" thick slab,}$$'%(t_concrete)
txt += (r'$$\text{the %s, %s, 4x4 joists need to be spaced at %g" on center,}$$'
        %(Grade,Species,joist_spacing.value.inches))
txt += (r'$$\text{the 4x6 stringers need to be spaced at %g" on center,}$$'
        %stringer_spacing.value.inches)
D = ceil(Shore_section_d)
B = ceil(Shore_section_b)
txt += r"$$\text{and the %g' tall %dx%d shores "%(Shore_length, B,D)
txt += r'need to be spaced at %g" on center.}$$'%actual_shore_spacing
summary = txt

In [53]:
Latex(summary)

<IPython.core.display.Latex object>

## References:

[Class website](https://ufl.instructure.com/courses/) *(Use this link to if you are taking the course on e-learning.)*

[Github.io version of course website](http://damontallen.github.io/Construction-Lectures-Fall-2014/) *(Do not use this link if you are taking this course in Summer A or B.)*

[IPython.org](http://ipython.org/) *(IPython is the opensource software used in the development of much of this course.)*

[Complete Software List](http://nbviewer.ipython.org/github/damontallen/Construction-Lectures-Fall-2014/blob/master/Resources.ipynb)